osu.Game.Tests/Visual/SongSelect/TestSceneBeatmapMetadataDisplay.cs

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using NUnit.Framework;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Utils;
+using osu.Game.Beatmaps;
+using osu.Game.Rulesets;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Screens.Play;
+using osuTK;
+
+namespace osu.Game.Tests.Visual.SongSelect
+{
+    public class TestSceneBeatmapMetadataDisplay : OsuTestScene
+    {
+        private BeatmapMetadataDisplay display;
+
+        [Resolved]
+        private BeatmapManager manager { get; set; }
+
+        [Resolved]
+        private RulesetStore rulesets { get; set; }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            foreach (var ruleset in rulesets.AvailableRulesets)
+                AddStep($"switch to {ruleset.Name}", () => Ruleset.Value = ruleset);
+        }
+
+        private void createTest(Func<WorkingBeatmap> getBeatmap)
+        {
+            AddStep("setup display", () => Child = display = new BeatmapMetadataDisplay(getBeatmap(), new Bindable<IReadOnlyList<Mod>>(getRandomMods()), Empty())
+            {
+                Anchor = Anchor.Centre,
+                Origin = Anchor.Centre,
+                Scale = new Vector2(1.5f),
+                Alpha = 0f,
+            });
+
+            AddStep("fade in", () => display.FadeIn(400, Easing.OutQuint));
+            AddToggleStep("trigger loading", v => display.Loading = v);
+        }
+
+        [Test]
+        public void TestLocal([Values("Beatmap", "Some long title and stuff")]
+                              string title,
+                              [Values("Trial", "Some1's very hardest difficulty")]
+                              string version)
+        {
+            createTest(() => CreateWorkingBeatmap(new Beatmap
+            {
+                BeatmapInfo =
+                {
+                    Metadata = new BeatmapMetadata
+                    {
+                        Title = title,
+                    },
+                    Version = version,
+                    StarDifficulty = RNG.NextDouble(0, 10),
+                }
+            }));
+        }
+
+        [Test]
+        public void TestRandomFromDatabase()
```

Comment:
really not sure about adding more test scenes which depend on the local beatmaps to function correctly

Reply:
added it to check how metadata would look on real local beatmaps, and I imagined it to be useful for others (especially when it comes to tweaking the design).

unless you think it's better not to rely on local databases all the time and always go with the option of adding programmatically written beatmaps.